flow-decomposition/src/main/java/com/powsybl/flow_decomposition/NetworkUtil.java

```diff
@@ -83,11 +81,63 @@ private static boolean isInMainSynchronousComponent(Branch<?> branch) {
             && isTerminalInMainSynchronousComponent(branch.getTerminal2());
     }
 
-    static boolean isTerminalInMainSynchronousComponent(Terminal terminal) {
+    private static boolean isTerminalInMainSynchronousComponent(Terminal terminal) {
         return terminal.getBusBreakerView().getBus().isInMainSynchronousComponent();
     }
 
-    static String getXnecId(String contingencyId, String branchId) {
-        return contingencyId.isEmpty() ? branchId : String.format("%s_%s", branchId, contingencyId);
+    public static List<Injection<?>> getNodeList(Network network) {
+        return getAllNetworkInjections(network)
+            .filter(NetworkUtil::isNotPairedDanglingLine)
+            .filter(NetworkUtil::isInjectionConnected)
+            .filter(NetworkUtil::isInjectionInMainSynchronousComponent)
+            .filter(NetworkUtil::managedInjectionTypes)
+            .toList();
+    }
+
+    public static List<Injection<?>> getXNodeList(Network network) {
+        return network.getDanglingLineStream()
+            .filter(dl -> !dl.isPaired())
+            .filter(NetworkUtil::isInjectionConnected)
+            .filter(NetworkUtil::isInjectionInMainSynchronousComponent)
+            .map(danglingLine -> (Injection<?>) danglingLine)
+            .collect(Collectors.toList());
+    }
+
+    private static boolean managedInjectionTypes(Injection<?> injection) {
+        return !(injection instanceof BusbarSection || injection instanceof ShuntCompensator || injection instanceof StaticVarCompensator); // TODO Remove this fix once the active power computation after a DC load flow is fixed in OLF
+    }
+
+    private static Stream<Injection<?>> getAllNetworkInjections(Network network) {
+        return network.getConnectableStream()
+            .filter(Injection.class::isInstance)
+            .map(connectable -> (Injection<?>) connectable);
+    }
+
+    private static boolean isInjectionConnected(Injection<?> injection) {
+        return injection.getTerminal().isConnected();
+    }
+
+    private static boolean isNotPairedDanglingLine(Injection<?> injection) {
+        return !(injection instanceof DanglingLine danglingLine && danglingLine.isPaired());
+    }
+
+    private static boolean isInjectionInMainSynchronousComponent(Injection<?> injection) {
+        return NetworkUtil.isTerminalInMainSynchronousComponent(injection.getTerminal());
+    }
+
+    public static List<String> getPstIdList(Network network) {
+        return network.getTwoWindingsTransformerStream()
+            .filter(NetworkUtil::isPst)
```

Comment:
```suggestion
            .filter(PhaseTapChangerHolder::hasPhaseTapChanger)
```

And remove the `isPst` function